builder/buildable.go

```diff
@@ -38,33 +38,28 @@ func NewBuildable(client client.Client,
 	}
 }
 
-func (buildable *Buildable) GetWorkflowBuild(req ctrl.Request, workflowID string) (apiv08.KogitoServerlessBuild, error) {
-	buildInstance := &apiv08.KogitoServerlessBuild{}
-	buildInstance.Spec.WorkflowId = workflowID
-	error := buildable.Client.Get(buildable.Ctx, req.NamespacedName, buildInstance)
-	return *buildInstance, error
+// GetWorkflowBuild gets the required Build associated with the same name/namespaced as defined in the request, nil if not found
+func (buildable *Buildable) GetWorkflowBuild(name string, namespace string) (*operatorapi.KogitoServerlessBuild, error) {
+	buildInstance := &operatorapi.KogitoServerlessBuild{}
+	err := buildable.Client.Get(buildable.Ctx, types.NamespacedName{Namespace: namespace, Name: name}, buildInstance)
+	if err != nil && k8serrors.IsNotFound(err) {
+		return nil, nil
+	}
+	return buildInstance, err
 }
 
-func (buildable *Buildable) CreateWorkflowBuild(workflowID string, targetNamespace string) (apiv08.KogitoServerlessBuild, error) {
-	buildInstance := &apiv08.KogitoServerlessBuild{}
+func (buildable *Buildable) getWorkflowBuild(req ctrl.Request, workflowID string) (*operatorapi.KogitoServerlessBuild, error) {
+	buildInstance := &operatorapi.KogitoServerlessBuild{}
 	buildInstance.Spec.WorkflowId = workflowID
-	buildInstance.ObjectMeta.Namespace = targetNamespace
-	buildInstance.ObjectMeta.Name = workflowID
-	error := buildable.Client.Create(buildable.Ctx, buildInstance)
-	return *buildInstance, error
+	err := buildable.Client.Get(buildable.Ctx, req.NamespacedName, buildInstance)
+	return buildInstance, err
 }
 
-func (buildable *Buildable) HandleWorkflowBuild(workflowID string, req ctrl.Request) (apiv08.KogitoServerlessBuild, error) {
-	buildInstance, error := buildable.GetWorkflowBuild(req, workflowID)
-	if error != nil {
-		if k8serrors.IsNotFound(error) {
-			return buildable.CreateWorkflowBuild(workflowID, req.Namespace)
-		}
-		// Error reading the object - requeue the request.
-		log.Error(error, "Failed to get KogitoServerlessBuild")
-		return buildInstance, error
-
-	} else {
-		return buildInstance, nil
-	}
+func (buildable *Buildable) CreateWorkflowBuild(workflowName string, targetNamespace string) (*operatorapi.KogitoServerlessBuild, error) {
+	buildInstance := &operatorapi.KogitoServerlessBuild{}
+	buildInstance.Spec.WorkflowId = workflowName
```

Comment:
Better to add a comment that the previous unique ID is considered now as the workflow name

Reply:
Same thing